src/new/wallet.js

```diff
@@ -1330,6 +1330,14 @@ class HathorWallet extends EventEmitter {
     this.emit('state', state);
   }
 
+  /**
+   * Enqueue the call for onNewTx with the given data.
+   * @param {{ history: import('../types').IHistoryTx }} wsData
+   */
+  enqueueOnNewTx(wsData) {
+    this.newTxPromise = this.newTxPromise.then(() => this.onNewTx(wsData));
```

Comment:
question: For most cases I believe this should be perfectly fine. However, if I understand correctly, every new transaction will preserve the last promise and add a new one.

Would this be a significant memory leak for wallets with a large number of transactions?